src/lib/utils.test.ts

```diff
@@ -720,6 +720,168 @@ describe('utils', () => {
     });
   });
 
+  describe('encodeData with custom array length and starting index', () => {
+    const schemas: ERC725JSONSchema[] = [
+      {
+        name: 'AddressPermissions[]',
+        key: '0xdf30dba06db6a30e65354d9a64c609861f089545ca58c6b4dbe31a5f338cb0e3',
+        keyType: 'Array',
+        valueType: 'address',
+        valueContent: 'Address',
+      },
+    ];
+
+    it('should be able to specify the array length + starting index', () => {
+      const encodedArraySection = encodeData(
+        [
+          {
+            keyName: 'AddressPermissions[]',
+            value: [
+              '0x983abc616f2442bab7a917e6bb8660df8b01f3bf',
+              '0x56ecbc104136d00eb37aa0dce60e075f10292d81',
+            ],
+            arrayLength: 23,
```

Comment:
I have the feeling this can be a bit flaky. Like ppl can mess up a bit with this.

What about changing this to "currentArrayLength"

And then, the value you return is `currentArrayLength + value.length`

Then there is no way to have a wrong number, or at least it is harder to fail it.


Because right now, if let's say:

- current array of size = 5
- you add 3 values
- you give `arrayLength=4`

You have a valid (as per your code) values.length > arrayLength. But you have a wrong value of array length = 4 but it should be 5 + 3 = 8.

I think this suggestion might make it less error prone.

what do you think?

Reply:
For next time, I suggest to open issues and have the discussion on the issue page so everyone (incl. me and external Devs) can get the whole discussion 😉

Reply:
Then maybe you can call it "finalArrayLength"?

Or something. Just as arrayLength can be confusing. Is it before? After? etc.